tools/src/format.rs

```diff
@@ -111,6 +145,25 @@ pub fn format_code(jumps: &HashMap<String, Jump>, format: FormatAs,
 
         let data = match (&token.kind, datum) {
             (&tokenize::TokenKind::Identifier(None), Some(d)) => {
+                // If this symbol starts a relevant nesting range and we haven't already pushed a
+                // symbol for this line, push it onto our stack.  A range is "relevant" if it
+                // its end line is not on the current line or the next line and therefore will
+                // actually trigger the "position:sticky" display scenario.
+                for a in d.iter() {
+                    let nests = match (a.nesting_range.start_lineno, nesting_stack.last()) {
+                        (0, _) => false,
+                        (_, None) => true,
+                        (a_start, Some(top)) if a_start == cur_line &&
```

Comment:
This might be a bit clean to write as:

```rust
(a_start, Some(top)) => {
  a_start == cur_line &&
  a_start != top.nesting_range.start_lineno &&
  a.nesting_range.end_lineno > cur_line + 1
}

Reply:
Nice cleanup, thanks!  I've spent too long away from my scheme/lisp roots! ;)